### PR TITLE
fix: docker uses virtualenv to avoid pip vs ubuntu problems

### PR DESCRIPTION
**Pull request by william-silversmith**

### CODE REVIEW

**Comment by tartavull** [comment]
the problem this tries to solve is that when you do `pip install requirements.txt` sometimes pypi server will timeout, and the requests version bundle with ubuntu doesn't correctly handle the retries?

The packages we are explicitly install using apt-get are h5py because it fails to install otherwise will still be available inside the virtualenv.

I find a bit confusing the need for a virtual enviroment inside a docker image, because the image already provides the isolation we want.

**Comment by tartavull** [comment]
http://stackoverflow.com/questions/29146792/why-people-create-virtualenv-in-a-docker-container
here are some answers to my questions

File: python/Dockerfile [hunk 1]
```diff
@@ -21,6 +21,8 @@ RUN sed -i -- 's/\"disable_updater\": false/\"disable_updater\": true/g' /google
 RUN mkdir /.ssh
 ENV PATH /google-cloud-sdk/bin:$PATH
 ADD ./ /root/python
+RUN pip install virtualenv virtualenvwrapper
+RUN bash /root/python/venv/bin/activate
```
Comment (tartavull) [inline python/Dockerfile]: I'm still confused about this bash command.
when we use this in production are the env variables for the virtual environment going to be set?

**Comment by william-silversmith** [comment]
This needs to be rethought

**Pull request closed without merging**